src/relax/op/ccl/ccl.cc

```diff
@@ -50,6 +50,44 @@ TVM_REGISTER_OP("relax.ccl.allreduce")
     .set_attr<FRelaxInferLayout>("FRelaxInferLayout", InferLayoutUnaryEwise)
     .set_attr<Bool>("FPurity", Bool(true));
 
+/* relax.ccl.allgather */
+TVM_REGISTER_NODE_TYPE(AllGatherAttrs);
+
+Expr allgather(Expr x, int num_workers) {
+  ObjectPtr<AllGatherAttrs> attrs = make_object<AllGatherAttrs>();
+  attrs->num_workers = std::move(num_workers);
+  static const Op& op = Op::Get("relax.ccl.allgather");
+  return Call(op, {std::move(x)}, Attrs{attrs}, {});
+}
+
+TVM_REGISTER_GLOBAL("relax.op.ccl.allgather").set_body_typed(allgather);
+
+StructInfo InferStructInfoAllGather(const Call& call, const BlockBuilder& ctx) {
+  TensorStructInfo input_sinfo = GetUnaryInputTensorStructInfo(call, ctx);
```

Comment:
With `num_workers` as a `PrimValue`, this would instead be 

```c++
CHECK_EQ(call->args.size(), 2);
auto input_sinfo = Downcast<TensorStructInfo>(call->args[0]->struct_info_);
auto num_workers_sinfo = Downcast<PrimStructInfo>(call->args[1]->struct_info_);

auto num_workers = num_workers_sinfo->value;
```